src/SessionMonitor/index.jsx

```diff
@@ -0,0 +1,56 @@
+import { userapiPath } from '../localconf';
+import getReduxStore from '../reduxStore';
+import { logoutAPI } from '../actions';
+
+/* eslint-disable class-methods-use-this */
+class SessionMonitor {
+  constructor(updateSessionTime, inactiveTimeLimit) {
+    this.updateSessionTime = updateSessionTime || 10 * 60 * 1000;
+    this.inactiveTimeLimit = inactiveTimeLimit || 30 * 60 * 1000;
+    this.mostRecentActivityTimestamp = Date.now();
+    this.interval = null;
+  }
+
+  start() {
+    window.addEventListener('mousedown', () => this.updateUserActivity(), false);
+    window.addEventListener('keypress', () => this.updateUserActivity(), false);
+    this.interval = setInterval(
+      () => this.updateSession(),
+      this.updateSessionTime,
+    ); // check session every X min
+  }
+
+  stop() {
+    clearInterval(this.interval);
+    window.removeEventListener('mousedown', this.updateUserActivity(), false);
+    window.removeEventListener('keypress', this.updateUserActivity(), false);
+  }
+
+  updateUserActivity() {
+    this.mostRecentActivityTimestamp = Date.now();
+  }
+
+  updateSession() {
+    // If user has been inactive for Y min
+    if (Date.now() - this.mostRecentActivityTimestamp >= this.inactiveTimeLimit) {
+      this.logoutUser();
+    } else if (Date.now() - this.mostRecentActivityTimestamp < this.inactiveTimeLimit) {
+      this.refreshSession();
+    } else {
+      console.log('Error calculating inactive time');
+    }
+  }
+
+  refreshSession() {
+    // hitting Fence endpoint refreshes token
+    fetch(userapiPath);
+  }
+
+  logoutUser() {
+    getReduxStore().then((store) => {
+      store.dispatch(logoutAPI());
+    });
+  }
+}
+
+export default SessionMonitor;
```

Comment:
Since we only ever want one SessionManager running - you can implement a `singleton` pattern here:
   https://en.wikipedia.org/wiki/Singleton_pattern
```
const singleton = new SessionMonitor();
export default singleton;
```
then the client does something like:
```
import sessionMonitor from ...;
sessionMonitor.start();
```